RecoHGCal/TICL/plugins/TrackstersProducer.cc

```diff
@@ -21,19 +21,26 @@
 #include "DataFormats/HGCalReco/interface/TICLSeedingRegion.h"
 
 #include "RecoHGCal/TICL/interface/PatternRecognitionAlgoBase.h"
+#include "RecoHGCal/TICL/interface/GlobalCache.h"
 #include "PatternRecognitionbyCA.h"
 #include "PatternRecognitionbyMultiClusters.h"
 
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
 using namespace ticl;
 
-class TrackstersProducer : public edm::stream::EDProducer<> {
+class TrackstersProducer : public edm::stream::EDProducer<edm::GlobalCache<TrackstersCache>> {
```

Comment:
this patterns seems to make the ticl::TrackstersCache -> ticl::CacheBase hierarchy useless.
There is no way to select a different algo without changing the producer as well.
Shouldn't the  CacheBase be removed?

Reply:
I interpreted the hard-coded initializer

```cpp
myAlgo_(std::make_unique<PatternRecognitionbyCA>(ps, cache))
```

as a temporary decision, given that `myAlgo_` refers to a generic `PatternRecognitionAlgoBase` ptr.

Regarding the TrackstersCache -> CacheBase structure, my intention here was that I didn't want to force all possible (future) algos derived from that base class to explicitly expect a specialized `TrackstersCache`.